ksqldb-engine/src/main/java/io/confluent/ksql/physical/scalablepush/PushRouting.java

```diff
@@ -98,18 +132,34 @@ public void close() {
       final ServiceContext serviceContext,
       final PushPhysicalPlan pushPhysicalPlan,
       final ConfiguredStatement<Query> statement,
-      final List<KsqlNode> hosts,
+      final Collection<KsqlNode> hosts,
       final LogicalSchema outputSchema,
-      final TransientQueryQueue transientQueryQueue
+      final TransientQueryQueue transientQueryQueue,
+      final PushConnectionsHandle pushConnectionsHandle,
+      final boolean dynamicallyAddedNode
   ) {
     final Map<KsqlNode, CompletableFuture<RoutingResult>> futureMap = new LinkedHashMap<>();
-    final CompletableFuture<Void> errorCallback = new CompletableFuture<>();
     for (final KsqlNode node : hosts) {
+      pushConnectionsHandle.add(
+          node, new RoutingResult(RoutingResultStatus.IN_PROGRESS, () -> { }));
+      final CompletableFuture<Void> callback = new CompletableFuture<>();
+      callback.handle((v, t) -> {
+        if (t == null) {
+          pushConnectionsHandle.get(node)
+              .ifPresent(result -> {
+                result.close();
+                result.updateStatus(RoutingResultStatus.COMPLETE);
+              });
+          LOG.info("Host {} completed request.", node);
```

Comment:
Yes, I added pushPhysicalPlan.getQueryId() to most of the logging calls to make it a bit easier to understand, though we don't log every request because it could result in a lot of logging.  They probably happen less often than pull queries, but we want them to be that lightweight.

I still think there's some value having semi exceptional or uncommon things in the logs to at least know something even if we don't know how to tie it to an exact query.  This usually should only happen if a node is shut down which doesn't happen often.